core/src/main/java/com/graphhopper/routing/PathExtract.java

```diff
@@ -0,0 +1,100 @@
+package com.graphhopper.routing;
+
+import com.graphhopper.routing.bwdcompat.AnnotationAccessor;
+import com.graphhopper.routing.bwdcompat.BoolAccessor;
+import com.graphhopper.routing.bwdcompat.SpeedAccessor;
+import com.graphhopper.routing.profiles.BitEncodedValue;
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.profiles.EncodingManager;
+import com.graphhopper.routing.util.DataFlagEncoder;
+import com.graphhopper.routing.util.EncodingManager08;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.util.*;
+
+public class PathExtract {
+    private Path path;
+    private SpeedAccessor speedAccessor;
+    private BoolAccessor roundaboutAccess;
+
+    PathExtract(Path path, EncodingManager08 encodingManager) {
+        this.path = path;
+
+        try {
+            final FlagEncoder encoder = path.weighting.getFlagEncoder();
+            if (encoder instanceof DataFlagEncoder) {
+                final DataFlagEncoder dataFlagEncoder = (DataFlagEncoder) encoder;
+                speedAccessor = new SpeedAccessor() {
+                    public double getSpeed(EdgeIteratorState edge) {
+                        return dataFlagEncoder.getMaxspeed(edge, 0, false);
+                    }
+                };
+            } else {
+                speedAccessor = new SpeedAccessor() {
+                    public double getSpeed(EdgeIteratorState edge) {
+                        return encoder.getSpeed(edge.getFlags());
+                    }
+                };
+            }
+
+            roundaboutAccess = new BoolAccessor() {
+                @Override
+                public boolean get(EdgeIteratorState edge) {
+                    return encoder.isBool(edge.getFlags(), FlagEncoder.K_ROUNDABOUT);
+                }
+            };
+        } catch (Exception ex) {
```

Comment:
Do not care about it this much. It will&must disappear. Just wanted to make this backward compatible and working